piq/kid.py

```diff
@@ -29,13 +29,11 @@ def _polynomial_kernel(X: torch.Tensor, Y: torch.Tensor = None, degree: int = 3,
     if Y is None:
         Y = X
 
-    if X.dim() != 2 or Y.dim() != 2:
-        raise ValueError('Incompatible dimension for X and Y matrices: '
-                         'X.dim() == {} while Y.dim() == {}'.format(X.dim(), Y.dim()))
+    assert X.dim() == 2 and Y.dim() == 2,\
+        f'Incompatible dimension for X and Y matrices: X.dim() = {X.dim()}, while Y.dim() = {Y.dim()}'
 
-    if X.size(1) != Y.size(1):
-        raise ValueError('Incompatible dimension for X and Y matrices: '
-                         'X.size(1) == {} while Y.size(1) == {}'.format(X.size(1), Y.size(1)))
+    assert X.size(1) == Y.size(1), \
+        f'Incompatible dimension for X and Y matrices: X.size(1) == {X.size(1)} while Y.size(1) == {Y.size(1)}'
 
```

Comment:
Indeed, matrix multiplication at line 41 fails in case of different size of the matrices. I would suggest to keep our validation due to better explanation of error compared to exceptions of `torch.mm`.
Tensor with more or less than 2 dimensions:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: mat2 must be a matrix
```
Tensors that can't be multiplied due to their shapes:
```
>>> torch.mm(a,b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: mat1 and mat2 shapes cannot be multiplied (2x3 and 2x3)
```